### DISCUSSION TITLE
Starcoin 主网启动指南｜Starcoin Main network launch Guide

### DISCUSSION THREAD

**Discussion by jolestar** [starcoinorg/starcoin#2506] [General]
Starcoin 主网将于 Barnard 网络 310000 高度启动，届时全球用户可以一起启动 Starcoin 主网。当 Barnard 网络到达 310000 高度时，主网节点会等待 6 个区块确认后，以 Barnard 网络 310000 高度区块的哈希作为主网创始块的父哈希，生成创世区块。 当主网启动后，开发团队会第一时间公布创世块哈希，将创世块固化到节点程序中，发布 Starcoin v1.0 正式版本。

### 下载 & 安装

在 [Release](https://github.com/starcoinorg/starcoin/releases/) 页面下载最新版本的 Starcoin 节点的二进制包，当前最新版本是 [v1.0.0-rc ](https://github.com/starcoinorg/starcoin/releases/tag/v1.0.0-rc)，或者用 docker 镜像 ：

```bash
docker pull starcoin/starcoin:v1.0.0-rc
```

### 启动节点加入主网

```bash
starcoin -n main --disable-miner-client true
```   
或者用 docker 启动

```bash
docker run --name starcoin-main -d --network host -v ~/.starcoin/:/root/.starcoin/ starcoin/starcoin:v1.0.0-rc /starcoin/starcoin -n main  --disable-miner-client true
```

> 注：当前主网节点启动后，会一直持续打印如下日志，等待 Barnard 的区块
>
> 2021-05-14T17:17:22.728540+08:00 INFO - barnard's latest block number is 291213
>2021-05-14T17:17:22.809420+08:00 INFO - Waiting to barnard's block 310000, main network will launch at 2021-05-17 > 20:47:14.809082 UTC, local time: 2021-05-18 04:47:14.809083 +08:00, remaining  3 days 11 hours 29 minutes 52 seconds (The time is estimated according to the current block time target[16000 millisecond] of barnard network)

更多操作请参看文档 [加入网络](https://developer.starcoin.org/zh/setup/runnetwork/)

### 接入矿机挖矿

等 Barnard 高度达到 310000 后，主网启动，节点的端口才可用，可以链接矿机挖矿。具体文档参看：https://starcoin-miner.gitbook.io/starcoin-miner-guide/

-----------------------------

Starcoin mainnet will be activated on Barnard Network 310000 height, and the hash of the Barnard Network 310000 block will be used as the parent hash of the mainnet genesis block, the global users can start Starcoin mainnet together.

### Download & Installation

Download the latest version of the Starcoin node binary package from the [Release](https://github.com/starcoinorg/starcoin/releases/) page, the latest version is [v1.0.0-rc ](https://github.com/starcoinorg/starcoin/releases/tag/v1.0.0-rc), or use the docker image: 

```bash
docker pull starcoin/starcoin:v1.0.0-rc
```

### Start the node to join the main network

```bash
starcoin -n main --disable-miner-client true
```   
Or start it with docker

```bash
docker run --name starcoin-main -d --network host -v ~/.starcoin/:/root/.starcoin/ starcoin/starcoin:v1.0.0-rc /starcoin/ starcoin -n main --disable-miner-client true
```

> Note: After the current node start, it will keep printing the following log, waiting for Barnard's block
>
> 2021-05-14T17:17:22.728540+08:00 INFO - barnard's latest block number is 291213
>2021-05-14T17:17:22.809420+08:00 INFO - Waiting to barnard's block 310000, main network will launch at 2021-05-17 > 20:47: 14.809082 UTC, local time: 2021-05-18 04:47:14.809083 +08:00, remaining 3 days 11 hours 29 minutes 52 seconds (The time is estimated according to the current block time target [16000 millisecond] of barnard network)

### Connect mining machine to mine

After the Barnard height reaches 310000, the main network will start and the ports of the node will be available for mining. See the documentation at https://starcoin-miner.gitbook.io/starcoin-miner-guide/

### Links

* [Starcoin v1.0.0-rc](https://github.com/starcoinorg/starcoin/releases/tag/v1.0.0-rc)
* [区块浏览器 | Mainnet Block Explorer](https://explorer.starcoin.org/main)
* [挖矿指南 | Miner guide](https://starcoin-miner.gitbook.io/starcoin-miner-guide/)
* [开发文档 | Developer documents](https://developer.starcoin.org/)

## FAQ

1. 当前 Barnard 的测试网节点会自动切换吗？ 不会，主网需要运行一个新的节点。
2. 如何保留当前 Barnard 的私钥和地址？将 barnard data 目录（默认 ～/.starcoin/barnard）下的 account_vaults 目录复制到 main data 目录 （默认 ～/.starcoin/main）

**Comment by xaoyaolee** [comment]
只能矿机挖矿吗？是什么算法，在网站上找不到任何说明，是否可以GPU挖？有没有miner的代码？

**Reply by jolestar** [reply]
当前用的是一种 crypto-night https://github.com/starcoinorg/starcoin/blob/master/consensus/src/cn.rs

**Comment by mzcaptainyang** [comment]
```
starcoin -n main --disable-miner-client true 
```
个人的 Mac 电脑，启动了上面的命令，进入了 console，也创建了账号，如何把账号同步到主网，同时开启挖矿呢？下面的命令，好像也不起作用
```
starcoin_miner -n 4
```

**Reply by jolestar** [reply]
> ```
> starcoin -n main --disable-miner-client true 
> ```
> 
> 个人的 Mac 电脑，启动了上面的命令，进入了 console，也创建了账号，如何把账号同步到主网，同时开启挖矿呢？下面的命令，好像也不起作用
> 
> ```
> starcoin_miner -n 4
> ```

要等挖出来块才行。现在难度比较高，不容易挖出来。

**Comment by danqing-xjcc** [comment]
连接到节点进行挖矿
当本地启动了 starcoin node 时，我们可以运行如下命令，启动4个线程连接到本地节点进行挖矿。
starcoin_miner -n 4
你好，我是通过docker构建的starcoin，我应该去哪里运行starcoin_miner -n 4挖矿命令？？？

**Reply by jolestar** [reply]
starcoin_miner 命令在 docker 镜像的 /starcoin/starcoin_miner

**Reply by danqing-xjcc** [reply]
root@iZbp179a0jchqqo80svs4cZ:~# docker exec -it starcoin bash
root@iZbp179a0jchqqo80svs4cZ:/# cd starcoin/
root@iZbp179a0jchqqo80svs4cZ:/starcoin# ls
mpm  starcoin  starcoin_db_exporter  starcoin_faucet  starcoin_generator  starcoin_indexer  starcoin_miner  starcoin_txfactory
root@iZbp179a0jchqqo80svs4cZ:/starcoin# ./starcoin_miner 
error: The following required arguments were not provided:
    --user <user>

USAGE:
    starcoin_miner --server <server> --thread-num <thread-num> --user <user>

For more information try --help
root@iZbp179a0jchqqo80svs4cZ:/starcoin#
你好，请问这是为什么？--user该怎么填写，你的教程中并未说到这个事情

**Reply by jolestar** [reply]
> root@iZbp179a0jchqqo80svs4cZ:~# docker exec -it starcoin bash root@iZbp179a0jchqqo80svs4cZ:/# cd starcoin/ root@iZbp179a0jchqqo80svs4cZ:/starcoin# ls mpm starcoin starcoin_db_exporter starcoin_faucet starcoin_generator starcoin_indexer starcoin_miner starcoin_txfactory root@iZbp179a0jchqqo80svs4cZ:/starcoin# ./starcoin_miner error: The following required arguments were not provided: --user
> 
> USAGE: starcoin_miner --server --thread-num --user
> 
> For more information try --help root@iZbp179a0jchqqo80svs4cZ:/starcoin# 你好，请问这是为什么？--user该怎么填写，你的教程中并未说到这个事情

后来加的，如果连接的是本地节点，这个字段随便写，如果连接的是矿池，根据矿池的要求写。

**Reply by danqing-xjcc** [reply]
运行主网节点
docker run --name starcoin -d --network host -v ~/.starcoin/:/root/.starcoin/ starcoin/starcoin:latest /starcoin/starcoin -n main
你好，请教一下，如上，我连接的是主网节点，请问该如何填写该字段？

**Reply by jolestar** [reply]
随便填写

**Reply by danqing-xjcc** [reply]
> 随便填写

root@iZbp179a0jchqqo80svs4cZ:/starcoin# ./starcoin_miner --user danqing
2022-02-22T05:02:41.549246125+00:00 INFO - Registry service: starcoin_service_registry::bus::service::BusService
2022-02-22T05:02:41.551439665+00:00 INFO - Service starcoin_service_registry::service_registry::RegistryService start.
2022-02-22T05:02:41.554423918+00:00 INFO - starcoin_service_registry::service_registry::RegistryService service actor started
2022-02-22T05:02:41.554905183+00:00 INFO - Put shared by type: starcoin_config::miner_config::MinerClientConfig
2022-02-22T05:02:41.556213571+00:00 INFO - Registry service: starcoin_miner_client::stratum_client_service::StratumClientService
2022-02-22T05:02:41.558001272+00:00 INFO - Put shared by type: starcoin_miner_client::stratum_client::StratumJobClient
2022-02-22T05:02:41.559549320+00:00 INFO - Registry service: starcoin_miner_client::miner::MinerClientService<starcoin_miner_client::stratum_client::StratumJobClient>
2022-02-22T05:02:41.560811537+00:00 INFO - Service starcoin_service_registry::bus::service::BusService start.
2022-02-22T05:02:41.561545179+00:00 INFO - starcoin_service_registry::bus::service::BusService service actor started
2022-02-22T05:02:41.564276496+00:00 INFO - Service starcoin_miner_client::stratum_client_service::StratumClientService start.
2022-02-22T05:02:41.564989029+00:00 INFO - starcoin_miner_client::stratum_client_service::StratumClientService service actor started
你好，请问我这样运行，是否表示成功开始挖矿？挖矿的产出该如何进行查看？

**Reply by jolestar** [reply]
主网挖不出来，可以连接测试网试试，看账户余额，或者查看节点的日志。

**Reply by danqing-xjcc** [reply]
> 主网挖不出来，可以连接测试网试试，看账户余额，或者查看节点的日志。

好的，谢谢
